vulnerabilities/importers/nvd.py

```diff
@@ -0,0 +1,153 @@
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an 'AS IS' BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import gzip
+import json
+import dataclasses
+from dateutil import parser as dateparser
+from datetime import date
+
+import requests
+
+from vulnerabilities.data_source import DataSource
+from vulnerabilities.data_source import Reference
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import DataSourceConfiguration
+
+
+@dataclasses.dataclass
+class NVDDataSourceConfiguration(DataSourceConfiguration):
+    etags: dict
+
+
+BASE_URL = "https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-{}.json.gz"
+
+
+class NVDDataSource(DataSource):
+
+    CONFIG_CLASS = NVDDataSourceConfiguration
+
+    def updated_advisories(self):
+        years = [1]
+        current_year = date.today().year
+        # NVD json feeds start from 2002.
+        for year in range(2002, current_year + 1):
+            download_url = BASE_URL.format(year)
+            if self.create_etag(download_url):
+                data = self.fetch(download_url)
+                yield self.to_advisories(data)
+
+    @staticmethod
+    def fetch(url):
+        gz_file = requests.get(url)
+        data = gzip.decompress(gz_file.content)
+        return json.loads(data)
+
+    def to_advisories(self, nvd_data):
+        for cve_item in nvd_data["CVE_Items"]:
+            if self.is_outdated(cve_item):
+                continue
+
+            if self.related_to_hardware(cve_item):
+                continue
+
+            cve_id = cve_item["cve"]["CVE_data_meta"]["ID"]
+            references = self.extract_references(cve_item)
+            summary = self.extract_summary(cve_item)
+            yield Advisory(
+                cve_id=cve_id, summary=summary, vuln_references=references, impacted_package_urls=[]
+            )
+
+    @staticmethod
+    def extract_summary(cve_item):
+        # In 99% of cases len(cve_item['cve']['description']['description_data']) == 1 , so
+        # this usually returns  cve_item['cve']['description']['description_data'][0]['value']
+        # In the remaining 1% cases this returns the longest summary.
+        summaries = [desc["value"] for desc in cve_item["cve"]["description"]["description_data"]]
+        return max(summaries, key=len)
```

Comment:
What about joining them with '\n' to handle all the cases nicely?
```Python
summaries = (desc["value"] for desc in cve_item["cve"]["description"]["description_data"])
summaries = (s.strip() for s in summaries if s)
return "\n".join(s for s in summaries if s)
```

Reply:
Thanks your solution is better with a caveat, in such cases , the `Summary` would be the joined summaries
![image](https://user-images.githubusercontent.com/28975399/89893921-d8906180-dbf6-11ea-837b-838c40e8ba56.png)

It probably won't make sense when reading such cases. 

Btw I only found only 1 CVE from NVD 2019 which had multiple(2) summaries.

Reply:
Your call... Since this is only  one case in many... this is best to have one weird summary and all other being correct (so IMHO the join is still better ;) )

Reply:
There are bunch of them in older CVE records. Choosing the longest one is the way to go  IMHo